tests/OpenTK.Tests.Math/QuaternionTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Xunit;
+using System;
+using System.Collections.Generic;
+
+namespace OpenTK.Tests.Math
+{
+	/// <summary>
+	/// Generates Quaternion test data.
+	/// </summary>
+	public class QuaternionTestDataGenerator
+	{
+		/// <summary>
+		/// Returns the single axis test cases.
+		/// 1. param: rotation in euler angles
+		/// 2. param: expected result of xyz-component of quaternion
+		/// 3. param: test name (Don't found a working way how to pass this to xUnit runner). I let it here for test documentation
+		/// </summary>
+		/// <value>The single axis test cases.</value>
+		public static IEnumerable<object> SingleAxisTestCases()
+		{
+			yield return new object[] { new Vector3(1, 0, 0), Vector3.UnitX, "Rotate around x axis" };
+			yield return new object[] { new Vector3(0, 1, 0), Vector3.UnitY, "Rotate around y axis" };
+			yield return new object[] { new Vector3(0, 0, 1), Vector3.UnitZ, "Rotate around z axis" };
+		}
+
+		/// <summary>
+		/// Returns the single ToAxisAngle test cases.
+		/// 1. param: Quaternion which a definied value of xyz-component.
+		/// 2. param: expected result of xyz-component of quaternion
+		/// 3. param: test name (Don't found a working way how to pass this to xUnit runner). I let it here for test documentation
+		/// </summary>
+		/// <value>The single axis test cases.</value>
+		public static IEnumerable<object[]> ToAxisAngleTestCases()
+		{
+			yield return new object[] { new Quaternion(Vector3.UnitX, 0), Vector3.UnitX, "Rotate around x axis" };
+			yield return new object[] { new Quaternion(Vector3.UnitY, 0), Vector3.UnitY, "Rotate around y axis" };
+			yield return new object[] { new Quaternion(Vector3.UnitZ, 0), Vector3.UnitZ, "Rotate around z axis" };
+		}
+	}
+
+	/// <summary>
+	/// Provides some methods which helps to verify test results
+	/// </summary>
+	internal static class QuaternionTestHelper
```

Comment:
Break this out into its own file, preferrably under a relevantly named subdirectory. Perhaps "Helpers".